### ISSUE TITLE
Please update ECF to produce content that's considered signed by the most recent LTS release of Java

### ISSUE THREAD

**Issue by merks** [eclipse/ecf#17]
Most of the content at this most recent site is signed with a signature/certificate that is excluded in the latest Java LTS releases (11 and 17) such that the runtime treats them as unsigned:

https://download.eclipse.org/rt/ecf/3.14.34/site.p2/3.14.34.v20221105-0315

This problem analogous to the following problem report which has additional background:

https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/661

Here is the generated report for the above ECF repository:

https://download.eclipse.org/oomph/archive/reports-extra/ecf/download.eclipse.org/rt/ecf/3.14.34/site.p2/3.14.34.v20221105-0315/index.html

I expect you will likely need to force qualifier changes on all the ECF plugins in this list.  Anything you pull from the Platform is already fixed in their latest 4.26 I-Build.  Content you get from Orbit is fixed in this repo:

https://download.eclipse.org/tools/orbit/downloads/drops/S20221109014815/repository

The Platform really needs this for RC2 and so does SimRel!

Please let me know what can I do to help?!

**Comment by merks** [comment]
I'm not sure if this is the right approach, increment by 1 the micro version of each feature and bundle that needs it?

![image](https://user-images.githubusercontent.com/208716/202463679-57879a68-64a5-4b26-9c3e-35842092076a.png)

Also the target platform needs to be changed...

I'll leave it at this for now, pending your advice and your desire for assistance...

**Comment by scottslewis** [comment]
When is RC2 for Platform?   Like EMF ECF is considered a -1 project.   

I'm not sure if I have the time to do all the version increments (both MANIFEST.MF and pom.xml with tycho) for all plugins and features manually.  Are there any scripts or utilities to increment versions that I can borrow?  (minor/service section only I guess).

**Comment by merks** [comment]
The new PR should fix the remaining problem.  The Platform's RC2 is scheduled for next Wednesday.  I expect they'll appreciate having the ECF build available earlier.  I'll contribute the ECF repo to SimRel for you when it's available. 

Thanks so much for your responsiveness!!

**Comment by scottslewis** [comment]
Hi Ed.   I've merged the new PR and will produce another snapshot build in a short while...and probably look to release ECF 3.14.35 on Monday next.

One quick question:   I was under the impression that you modified all the ECF pom/manifest entries for the entire ECF sdk (i.e. all plugins and all features).   There were a lot of them ~145...so I didn't check for completeness yet.  If that's not what the 145 changes were could you let me know so that I don't have to look them all over?

Also...for my info for future:  How did you update all of them (with just service segment increment?).  I would like to have such a tool around if it's available to me...just for the future.

Thanks.  I'll complete build and post snapshot build here when done for your and my verification.

**Comment by merks** [comment]
I only modified the features and plugins that were reported to have bad signatures, i.e., those signed between 2019.01.01 and ~2021.04.01 roughly; after that a new sha2 certificate was issued.   So the rest of the plugins are fine as is without need for a minor increment.  I had no automated tool to do this, so it was kind of painful.  I did use some shell scripting to create the long list of files to edit from the text of the reported problem and then I used vi from the command line to edit them.  I thought I'd go blind though...

Oomph does have a versioning tool to help automate these types of things, but it's a fair bit of work to set it up initially...

**Comment by scottslewis** [comment]
Thanks Ed for doing the hard work on this.  I appreciate that.   

I've released ECF 3.14.37 with all fixes version increments, etc.  The repo is here:

https://download.eclipse.org/rt/ecf/3.14.37/site.p2/3.14.37.v20221119-2038/

In a moment, I'll contribute the new ECF filetransfer on the Platform issue.  https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/591

Closing this issue.
